src/Namicorn.test.ts

```diff
@@ -314,72 +353,134 @@ describe('ENS', () => {
     const namicorn = new Namicorn({ blockchain: { ens: true } });
     expect(namicorn.ens.network).toBe('mainnet');
-    expect(namicorn.ens.url).toBe('https://mainnet.infura.io/');
+    expect(namicorn.ens.url).toBe('https://mainnet.infura.io');
   });
 
   it('checks normalizeSource ens (boolean - false)', async () => {
-    const namicorn = new Namicorn({ blockchain: { ens: false } });
-    expect(namicorn.isSupportedDomain('hello.eth')).toBeFalsy()
-    expect(namicorn.ens).toBeUndefined()
+    const ens = new Ens({ network: 5 });
+    expect(ens.network).toBe('goerli');
+    expect(ens.url).toBe('https://goerli.infura.io');
+    expect(ens.isSupportedNetwork()).toBeFalsy();
   });
 
   it('checks normalizeSource ens (object) #1', async () => {
     const namicorn = new Namicorn({
       blockchain: { ens: { url: 'https://mainnet.infura.io' } },
     });
     expect(namicorn.ens.network).toBe('mainnet');
-    expect(namicorn.ens.url).toBe('https://mainnet.infura.io/');
+    expect(namicorn.ens.url).toBe('https://mainnet.infura.io');
   });
 
   it('checks normalizeSource ens (object) #2', async () => {
     const namicorn = new Namicorn({ blockchain: { ens: { network: 3 } } });
     expect(namicorn.ens.network).toBe('ropsten');
-    expect(namicorn.ens.url).toBe('https://ropsten.infura.io/');
+    expect(namicorn.ens.url).toBe('https://ropsten.infura.io');
+    expect(namicorn.ens.registryAddress).toBe('0x112234455c3a32fd11230c42e7bccd4a84e02010');
   });
 
   it('checks normalizeSource ens (object) #3', async () => {
     const namicorn = new Namicorn({
-      blockchain: { ens: { url: 'https://rinkeby.infura.io/' } },
+      blockchain: { ens: { url: 'https://rinkeby.infura.io' } },
     });
     expect(namicorn.ens.network).toBe('rinkeby');
-    expect(namicorn.ens.url).toBe('https://rinkeby.infura.io/');
+    expect(namicorn.ens.url).toBe('https://rinkeby.infura.io');
   });
 
   it('checks normalizeSource ens (object) #4', async () => {
     const namicorn = new Namicorn({
-      blockchain: { ens: { url: 'https://goerli.infura.io/', network: 5 } },
+      blockchain: { ens: { url: 'https://goerli.infura.io', network: 5 } },
     });
     expect(namicorn.ens.network).toBe('goerli');
-    expect(namicorn.ens.url).toBe('https://goerli.infura.io/');
+    expect(namicorn.ens.url).toBe('https://goerli.infura.io');
+    expect(namicorn.ens.registryAddress).toBeUndefined();
   });
 
   it('checks normalizeSource ens (object) #6', async () => {
-    expect(() => {
-      new Namicorn({ blockchain: { ens: { network: 7543 } } });
-    }).toThrowError()
+    expect(() => 
+      new Namicorn({ blockchain: { ens: { network: 7543 } } })
+    ).toThrow();
   });
 
   it('checks normalizeSource ens (object) #7', async () => {
-    expect(() => {
-      new Namicorn({ blockchain: { ens: { network: 'invalid' } } });
-    }).toThrowError()
+    expect(() => 
+      new Namicorn({ blockchain: { ens: { network: 'invalid' } } })
+    ).toThrow();
```

Comment:
throwing an error because network is custom but url is not specified. 
Didn't do that on zns side (the structure is a little bit different there and source.url would be undefined if the network is custom)